testing/internal/e2e/tests/static_with_vault/credential_store_test.go

```diff
@@ -138,14 +165,20 @@ func TestCliVaultCredentialStore(t *testing.T) {
 	require.NoError(t, err)
 
 	newSessionAuthorization := newSessionAuthorizationResult.Item
-	retrievedUser := fmt.Sprintf("%s", newSessionAuthorization.Credentials[0].Credential["username"])
-	retrievedKey := fmt.Sprintf("%s", newSessionAuthorization.Credentials[0].Credential["private_key"])
+	retrievedUser := newSessionAuthorization.Credentials[0].Credential["username"].(string)
```

Comment:
What do you think about adding a len check before doing the index lookup? This would avoid a panic if we get an unexpected result.
```suggestion
	require.Len(t, newSessionAuthorization.Credentials, 2)
	retrievedUser := newSessionAuthorization.Credentials[0].Credential["username"].(string)
```